### PR TITLE
Put driver files block

### PR DESCRIPTION
**Pull request by vkz-mlx**
Adding the RC.1.3.0 as a base version for MLX75322 SPI driver

### CODE REVIEW

File: Make-75322-raspi.mk [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Makefile for customizing build for 75322 (Raspberry Pi-pased compilation)
```
Comment (kravchenkoloznia) [inline Make-75322-raspi.mk]: typo `Pi-pased`

File: Make-75322.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Makefile for customizing build for 75322 (PC-pased compilation)
```
Comment (kravchenkoloznia) [inline Make-75322.mk]: typo `PC-pased`

File: Make-75322.mk [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# RELEASE
+# Name of the release to build in case of versioned_release
+RELEASE_MATURITY = RC
+RELEASE_MAJOR    = 1
+RELEASE_MINOR    = 3
+RELEASE_REVISION = 0
```
Comment (kravchenkoloznia) [inline Make-75322.mk]: It seems like it's duplicating release version from `Make-75322-raspi.mk`

File: Make-75322-raspi.mk (hunk 1 above)
Comment (kravchenkoloznia) [reply Make-75322-raspi.mk]: Based on description that two files: `Make-75322-raspi.mk` and `Make-75322.mk`; used to customize build for the same product. So why we need both of them?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+RST_DOCS = $(sort $(wildcard $(DOC_DIR)/*.rst))
+
+INCLUDES += $(patsubst %, -I%, $(SRC_DIRS))
+
+CPPFLAGS  = -Wall
+CPPFLAGS += -fpic
+CPPFLAGS += -std=c99
+CPPFLAGS += -c -g -ffunction-sections -fdata-sections -fms-extensions\
+	-O -Wall -W
```
Comment (kravchenkoloznia) [inline Makefile]: what is the reason to add `\` symbol instead of add another line with `CPPFLAGS +=`

File: doc/Makefile [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+endif
+
+.PHONY: help clean html dirhtml singlehtml pickle json htmlhelp qthelp devhelp epub latex latexpdf text man changes linkcheck doctest gettext
+
+help:
+	@$(ECHO) "Please use \`make <target>' where <target> is one of"
+	@$(ECHO) "  html       to make standalone HTML files"
+	@$(ECHO) "  singlehtml to make a single large HTML file"
+	@$(ECHO) "  latex      to make LaTeX files, you can set PAPER=a4 or PAPER=letter"
+	@$(ECHO) "  latexpdf   to make LaTeX files and run them through pdflatex"
```
Comment (kravchenkoloznia) [inline doc/Makefile]: `doxy` target is not mentioned in help

File: doc/Makefile [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+# rest documentation, we need to make sure we publish it at the same time
+DOXYGEN_INPUT = $(addsuffix /src, $(COMMON_COMPONENT_DIRS) $(CUSTOM_COMPONENT_DIRS)) $(PRODUCT_SRCDIR)
+# also parse any inc folders from submoduled libraries
+DOXYGEN_INPUT += $(wildcard $(addsuffix /inc/., $(COMMON_COMPONENT_DIRS) $(CUSTOM_COMPONENT_DIRS)) $(PRODUCT_SRCDIR))
+DOXYGEN_INPUT += $(addsuffix /src, $(LIBRARY_DIRS))
+DOXYGEN_INPUT += $(addsuffix /inc, $(LIBRARY_DIRS))
+.PHONY: doxy
+doxy: $(BUILDDIR)/html/doxygen
+	@$(ECHO) "** Running doxygen on source code, destination of doxygen build is the final html folder."
+	$(HIDE_CMD) DOXYGEN_INPUT="$(DOXYGEN_INPUT)" COMPONENT_FLAGS="$(COMPONENT_CPPFLAGS)" doxygen doxygen/doxyfile $(NO_OUTPUT)
```
Comment (kravchenkoloznia) [inline doc/Makefile:84]: do we really need all other stuff in that **Makefile** if only `doxy` target is used?

File: doc/source/conf.py [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+# SWCC Project documentation build configuration file, created by
+# sphinx-quickstart on Fri Aug  1 16:00:01 2014.
+#
+# This file is execfile()d with the current directory set to its containing dir.
+#
+# Note that not all possible configuration values are present in this
+# autogenerated file.
+#
+# All configuration values have a default; values that are commented out
+# serve to show the default.
```
Comment (kravchenkoloznia) [inline doc/source/conf.py]: That file maybe deleted if only doxygen documentation will be used.

File: pip-dependencies.txt [hunk 8]
```diff
@@ -0,0 +1,20 @@
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/swcc-pypi-release/simple
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/pypi-release/simple
+--extra-index-url https://pypi.org/simple
+
+# Pip packages needed in general
+mlx.warnings>=0.2.0
```
Comment (kravchenkoloznia) [inline pip-dependencies.txt]: Is it used in that project?

File: pip-dependencies.txt [hunk 9]
```diff
@@ -0,0 +1,20 @@
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/swcc-pypi-release/simple
+--extra-index-url https://dsl.melexis.com/artifactory/api/pypi/pypi-release/simple
+--extra-index-url https://pypi.org/simple
```
Comment (kravchenkoloznia) [inline pip-dependencies.txt]: If `mlx.warnings` plugin is not used, extra index url's to `dsl.melexis.com` is not needed

File: doc/doxygen/footer.html [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<!-- HTML footer for doxygen 1.8.3.1-->
+<!-- start footer part -->
+<!--BEGIN GENERATE_TREEVIEW-->
+<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
+  <ul>
+    $navpath
+    <li class="footer">
+    <img class="footer" src="../images/melexis-logo.png" alt="Melexis N.V," height="25px" hspace="10"/>
+	&copy; Copyright 2015 Melexis N.V. ---
```
Comment (dryodon) [inline doc/doxygen/footer.html]: 2015?

**Review by kravchenkoloznia — COMMENTED** [review]
Some small comments left.

**Comment by dryodon** [comment]
Please explain in readme on what systems you built the driver and what you needed to do to build it.
Stress that you're not sure nor guaranteeing that it builds on other systems.

File: pip-dependencies.txt (hunk 8 above)
Comment (vkz-mlx) [reply pip-dependencies.txt]: whole file removed as not required

File: doc/Makefile (hunk 6 above)
Comment (vkz-mlx) [reply doc/Makefile:84]: Significantly cleaned-up

File: Makefile (hunk 4 above)
Comment (vkz-mlx) [reply Makefile]: One-line assignment used

File: Make-75322.mk (hunk 3 above)
Comment (vkz-mlx) [reply Make-75322.mk]: Moved to root Makefile

File: Make-75322.mk (hunk 2 above)
Comment (vkz-mlx) [reply Make-75322.mk]: Fixed

File: Make-75322-raspi.mk (hunk 1 above)
Comment (vkz-mlx) [reply Make-75322-raspi.mk]: customization and further family growth. Each file - each platform.

File: doc/doxygen/footer.html (hunk 10 above)
Comment (vkz-mlx) [reply doc/doxygen/footer.html]: Updated

File: pip-dependencies.txt (hunk 9 above)
Comment (vkz-mlx) [reply pip-dependencies.txt]: removed as described above https://github.com/melexis/mlx75322-spi-driver/pull/1#discussion_r382599563

**Pull request merged**